src/Nethermind/Nethermind.Evm/VirtualMachine.cs

```diff
@@ -818,6 +819,11 @@ private CallResult ExecuteCall<TTracingInstructions>(EvmState vmState, ReadOnlyM
 #endif
             Instruction instruction = (Instruction)code[programCounter];
 
+            if (isCancelable && _txTracer.IsCancelled)
+            {
+                ThrowOperationCanceledException();
+            }
```

Comment:
How is that much faster than IsTracingInstructions?

Reply:
IsTracingInstructions == true comes with the additional load of copying every stack data change, memory copy etc